src/python/grpcio_testing/grpc_testing/__init__.py

```diff
@@ -0,0 +1,441 @@
+        under test.
+
+        This method is mutative; for each RPC made by the system under test
+        exactly one ChannelRpc will be returned from this method. If two calls
+        to this method are made by test code but only one RPC is invoked by the
+        system under test, the second call to this method will block
+        indefinitely.
+
+        Args:
+          service_name: An RPC service name (such as "my_package.MyService").
+          method_name: An RPC method name (such as "MyMethod").
+
+        Returns:
+          A ChannelRpc that allows test code to "play server" and interact with
+            the system under test.
+        """
+        raise NotImplementedError()
+
+
+class ServerRpc(six.with_metaclass(abc.ABCMeta)):
+    """Fixture for an in-progress RPC serviced by a system under test.
+
+    Enables users to "play a client" for such RPCs by supplying requests,
+    reading responses, and cancelling.
+    """
+
+    @abc.abstractmethod
+    def initial_metadata(self):
+        """Accesses the initial metadata emitted by the system under test.
+
+        This method blocks until the system under test has added initial
+        metadata to the RPC (or has provided one or more response messages or
+        has terminated the RPC, either of which will cause gRPC Python to
+        synthesize initial metadata for the RPC).
+
+        Returns:
+          The initial metadata for the RPC.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def add_requests_as_messages(self, requests):
+        """Accepts requests to be passed to the system under test.
+
+        Args:
+          requests: An iterable of deserialized request messages to be passed
+            to the system under test.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def take_response_as_message(self):
+        """Draws one of the responses added to the RPC by the system under test.
+
+        Successive calls to this method return responses in the same order in
+        which the system under test added them to the RPC.
+
+        Returns:
+          A deserialized response message added to the RPC by the system under
+            test.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def no_more_requests(self):
+        """Indicates the end of the RPC's request stream."""
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def cancel(self):
+        """Cancel's the RPC."""
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def terminate(self):
+        """Blocks until the system under test has terminated the RPC."""
+        raise NotImplementedError()
+
+
+class ServerFixture(six.with_metaclass(abc.ABCMeta)):
+    """Fixture for a server in which a system under test services RPCs."""
+
+    @abc.abstractmethod
+    def invoke_rpc_by_service_and_method_names(self, service_name, method_name,
```

Comment:
Similar to the client, what about moving the service name/method name to the rpc object.